### PR TITLE
Forward selector exception in on_error_resume_next + expose test utils

### PR DESCRIPTION
**Pull request by CorentinBT**

### CODE REVIEW

File: src/rpp/rpp/operators/switch_on_next.hpp [hunk 1]
```diff
@@ -62,8 +62,8 @@ namespace rpp::operators::details
 
         void on_error(const std::exception_ptr& err) const
         {
-            m_state->dispose();
             m_state->get_observer()->on_error(err);
+            m_state->dispose();
```
Comment (CorentinBT) [inline src/rpp/rpp/operators/switch_on_next.hpp]: I think dispose should happen after on_error/on_completed emissions, same thing for on_error_resume_next

Comment (victimsnino) [reply src/rpp/rpp/operators/switch_on_next.hpp]: Why? My point is that this state contains disposable for upstream, while observer is downstream. So, you firstly dispose upstream to correctly stop it and then forward value

Comment (CorentinBT) [reply src/rpp/rpp/operators/switch_on_next.hpp]: but then if one attaches a disposable callback he will first see the callback being called before the error emission?

Comment (victimsnino) [reply src/rpp/rpp/operators/switch_on_next.hpp]: If it comes from upstream - yes

Comment (victimsnino) [reply src/rpp/rpp/operators/switch_on_next.hpp]: But I don't know what is better in this case :)

Comment (victimsnino) [reply src/rpp/rpp/operators/switch_on_next.hpp]: OK, we can use your approach, but what about rest combining/merging operators in this case? They are using same approach, for example, merge

Comment (victimsnino) [reply src/rpp/rpp/operators/switch_on_next.hpp]: discussed in discord: let's do it as separate PR for ALL combining/merging operators

Comment (CorentinBT) [reply src/rpp/rpp/operators/switch_on_next.hpp]: done I reverted disposable changes for now until next PR

**Review by victimsnino — APPROVED** [review]

**Pull request merged**